src/util/simplify_expr_int.cpp

```diff
@@ -894,7 +895,9 @@ bool simplify_exprt::simplify_concatenation(exprt &expr)
       if(op.is_true() || op.is_false())
       {
         bool value=op.is_true();
-        op=constant_exprt(value?ID_1:ID_0, unsignedbv_typet(1));
+        op = constant_exprt(
+          value ? ID_1 : ID_0, bitvector_typet(expr.type().id(), 1));
```

Comment:
value should be const, or could be inlined here

Reply:
True, but unrelated to the fix here? Let me know if you insist, though.